testsuite/src/test/java/org/ironjacamar/eis/unit/EchoTestCase.java

```diff
@@ -0,0 +1,144 @@
+import org.ironjacamar.embedded.junit4.AllChecks;
+import org.ironjacamar.embedded.junit4.IronJacamar;
+import org.ironjacamar.embedded.junit4.PostCondition;
+import org.ironjacamar.embedded.junit4.PreCondition;
+import org.ironjacamar.rars.ResourceAdapterFactory;
+
+import javax.annotation.Resource;
+
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Echo test for the EIS test server
+ * 
+ * @author <a href="mailto:jesper.pedersen@ironjacamar.org">Jesper Pedersen</a>
+ */
+@RunWith(IronJacamar.class)
+@Configuration(full = true)
+@PreCondition(condition = AllChecks.class)
+@PostCondition(condition = AllChecks.class)
+public class EchoTestCase
+{
+   // --------------------------------------------------------------------------------||
+   // Class Members ------------------------------------------------------------------||
+   // --------------------------------------------------------------------------------||
+
+   private static Logger log = Logger.getLogger(EchoTestCase.class);
+
+   private static EISServer server = new EISServer();
+
+
+   /**
+    * The resource adapter
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 1)
+   private static ResourceAdapterArchive createResourceAdapter() throws Throwable
+   {
+      return ResourceAdapterFactory.createEchoRar();
+   }
+
+   /**
+    * The activation
+    * @throws Throwable In case of an error
+    */
+   @Deployment(order = 2)
+   private static ResourceAdaptersDescriptor createActivation() throws Throwable
+   {
+      return ResourceAdapterFactory.createEchoDeployment(null);
+   }
+   //-------------------------------------------------------------------------------------||
+   // Tests ------------------------------------------------------------------------------||
+   //-------------------------------------------------------------------------------------||
+
+   /**
+    * Start EIS server
+    * @throws Throwable In case of an error
+    */
+   @BeforeClass
+   public static void startEISServer() throws Throwable
+   {
+      server.setHandlerClassName(EchoHandler.class.getName());
+      server.setHost("localhost");
+      server.setPort(1400);
+      server.startup();
+   }
+
+   /**
+    * Stop EIS server
+    * @throws Throwable In case of an error
+    */
+   @BeforeClass
+   public static void stopEISServer() throws Throwable
+   {
+      server.shutdown();
+   }
+
+
+   @Resource(mappedName = "java:/eis/echo")
+   private EchoConnectionFactory connectionFactory;
+ 
+   /**
+    * Basic
+    * @exception Throwable Thrown if case of an error
+    */
+   @Test
+   public void testBasic() throws Throwable
+   {
+      assertNotNull(connectionFactory);
```

Comment:
Redo to use a socket connection directly to the EISServer - e.g. w/o a resource adapter